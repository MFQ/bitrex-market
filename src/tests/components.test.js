import React from 'react';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import Symbol from './../components/symbol';
import Market from './../components/market';
import Header from './../components/header';

import CurrencyView from './../components/currency-component/currency-view';
import CurrencyBookView from './../components/currency-component/currency-book-view';

configure({ adapter: new Adapter() });

describe('Symbol', () => {
  it('should render correctly', () => {
    const mockData = {
      bids: [],
      asks: []
    }
    const output = shallow(
      <CurrencyBookView data={ mockData } />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});

describe('CurrencyView', () => {
  it('should render correctly', () => {
    const BTCDATA = {
      "mid": "9336.55",
      "bid": "9336.5",
      "ask": "9336.6",
      "last_price": "9336.6",
      "low": "9228.0",
      "high": "9393.9",
      "volume": "16146.421390230002",
      "timestamp": "1525963063.0559313"
    }
    const output = shallow(
      <CurrencyView symbol="btcusd" data={BTCDATA} />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});


describe('Symbol', () => {
  it('should render correctly', () => {
    const output = shallow(
      <Symbol index="1" symbolObj={{ name: 'name'}} />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});

describe('Market', () => {
  it('should render correctly', () => {
    const output = shallow( <Market currency="btc" symbols={[]} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});


describe('Header', () => {
  it('should render correctly', () => {
    const output = shallow( <Header/> );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
