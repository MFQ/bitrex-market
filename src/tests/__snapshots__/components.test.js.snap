// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyView should render correctly 1`] = `
<Container
  tag="div"
>
  <Row
    tag="div"
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <svg
        fill="grey"
        height={200}
        viewBox="0 0 226.777 226.777"
        width={200}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M135.715 122.244c-2.614-1.31-8.437-3.074-15.368-3.533-6.934-.458-15.828 0-15.828 0v30.02s9.287.198 15.503-.26c6.21-.458 12.621-2.027 15.826-3.795 3.203-1.766 7.063-4.513 7.063-11.379 0-6.869-4.579-9.745-7.196-11.053zm-19.555-17.465c5.104-.197 10.532-1.373 14.453-3.532 3.925-2.158 6.148-5.557 6.02-10.66-.134-5.102-3.532-9.418-9.287-11.186-5.757-1.766-9.613-1.897-13.998-1.962-4.382-.064-8.83.328-8.83.328v27.012c.001 0 6.541.197 11.642 0z"
        />
        <path
          d="M113.413 0C50.777 0 0 50.776 0 113.413c0 62.636 50.777 113.413 113.413 113.413s113.411-50.777 113.411-113.413C226.824 50.776 176.049 0 113.413 0zm46.178 156.777c-8.44 5.887-17.465 6.935-21.455 7.456-1.969.259-5.342.532-8.959.744v22.738h-13.998v-22.37h-10.66v22.37H90.522v-22.37H62.987l2.877-16.812h8.371c2.814 0 3.989-.261 5.166-1.372 1.177-1.113 1.439-2.812 1.439-4.188V85.057c0-3.628-.295-4.61-1.963-6.473-1.668-1.867-5.591-2.112-7.8-2.112h-8.091V61.939h27.535V39.505h13.996v22.434h10.66V39.505h13.998v22.703c10.435.647 18.203 2.635 24.983 7.645 8.766 6.475 8.306 17.724 8.11 20.406-.195 2.682-1.372 7.85-3.729 11.183-2.352 3.337-8.108 6.673-8.108 6.673s6.801 1.438 11.578 5.036c4.771 3.598 8.307 9.941 8.106 19.229-.192 9.288-2.088 18.511-10.524 24.397z"
        />
      </svg>
    </Col>
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Row
        key="0"
        tag="div"
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Row
            tag="div"
          >
            <strong>
              MID
            </strong>
          </Row>
        </Col>
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Row
            tag="div"
          >
            9336.55
          </Row>
        </Col>
      </Row>
      <Row
        key="1"
        tag="div"
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Row
            tag="div"
          >
            <strong>
              BID
            </strong>
          </Row>
        </Col>
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Row
            tag="div"
          >
            9336.5
          </Row>
        </Col>
      </Row>
      <Row
        key="2"
        tag="div"
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Row
            tag="div"
          >
            <strong>
              ASK
            </strong>
          </Row>
        </Col>
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Row
            tag="div"
          >
            9336.6
          </Row>
        </Col>
      </Row>
      <Row
        key="3"
        tag="div"
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Row
            tag="div"
          >
            <strong>
              LAST_PRICE
            </strong>
          </Row>
        </Col>
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Row
            tag="div"
          >
            9336.6
          </Row>
        </Col>
      </Row>
      <Row
        key="4"
        tag="div"
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Row
            tag="div"
          >
            <strong>
              LOW
            </strong>
          </Row>
        </Col>
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Row
            tag="div"
          >
            9228.0
          </Row>
        </Col>
      </Row>
      <Row
        key="5"
        tag="div"
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Row
            tag="div"
          >
            <strong>
              HIGH
            </strong>
          </Row>
        </Col>
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Row
            tag="div"
          >
            9393.9
          </Row>
        </Col>
      </Row>
      <Row
        key="6"
        tag="div"
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Row
            tag="div"
          >
            <strong>
              VOLUME
            </strong>
          </Row>
        </Col>
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Row
            tag="div"
          >
            16146.421390230002
          </Row>
        </Col>
      </Row>
      <Row
        key="7"
        tag="div"
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Row
            tag="div"
          >
            <strong>
              TIMESTAMP
            </strong>
          </Row>
        </Col>
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Row
            tag="div"
          >
            1525963063.0559313
          </Row>
        </Col>
      </Row>
    </Col>
  </Row>
</Container>
`;

exports[`Header should render correctly 1`] = `
<header>
  <Navbar
    color="dark"
    dark={true}
    expand="md"
    tag="nav"
  >
    <NavbarBrand
      href="/"
      tag="a"
    >
       Bitrex Market 
    </NavbarBrand>
  </Navbar>
</header>
`;

exports[`Market should render correctly 1`] = `
<div>
  <Alert
    closeAriaLabel="Close"
    color="secondary"
    isOpen={true}
    tag="div"
    transition={
      Object {
        "appear": true,
        "baseClass": "fade",
        "baseClassActive": "show",
        "enter": true,
        "exit": true,
        "in": true,
        "mountOnEnter": false,
        "onEnter": [Function],
        "onEntered": [Function],
        "onEntering": [Function],
        "onExit": [Function],
        "onExited": [Function],
        "onExiting": [Function],
        "tag": "div",
        "timeout": 150,
        "unmountOnExit": true,
      }
    }
  >
    <Row
      tag="div"
    >
       
      btc
       Markets. 
    </Row>
  </Alert>
  <Container
    tag="div"
  >
    <Row
      tag="div"
    >
      <Table
        responsiveTag="div"
        tag="table"
      >
        <thead>
          <tr>
            <th>
              #
            </th>
            <th>
               Action 
            </th>
          </tr>
        </thead>
        <tbody />
      </Table>
    </Row>
    <Row
      tag="div"
    >
       
    </Row>
  </Container>
</div>
`;

exports[`Symbol should render correctly 1`] = `
<Container
  tag="div"
>
  <Row
    tag="div"
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <div>
        <h1>
           
          Bids
           
        </h1>
        <Table
          responsiveTag="div"
          tag="table"
        >
          <thead>
            <tr>
              <th
                key="0"
              >
                 
                Amount
                 
              </th>
              <th
                key="1"
              >
                 
                Price
                 
              </th>
              <th
                key="2"
              >
                 
                Timestamp
                 
              </th>
            </tr>
          </thead>
          <tbody />
        </Table>
      </div>
    </Col>
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <div>
        <h1>
           
          Asks
           
        </h1>
        <Table
          responsiveTag="div"
          tag="table"
        >
          <thead>
            <tr>
              <th
                key="0"
              >
                 
                Amount
                 
              </th>
              <th
                key="1"
              >
                 
                Price
                 
              </th>
              <th
                key="2"
              >
                 
                Timestamp
                 
              </th>
            </tr>
          </thead>
          <tbody />
        </Table>
      </div>
    </Col>
  </Row>
</Container>
`;

exports[`Symbol should render correctly 2`] = `
<tr>
  <th
    scope="row"
  >
    1
  </th>
  <td
    key="0"
  >
     
    name
     
  </td>
  <td>
    <Link
      replace={false}
      to="/currencies/undefined"
    >
      <Button
        color="success"
        tag="button"
      >
         Detail 
      </Button>
    </Link>
  </td>
</tr>
`;
